RecoParticleFlow/PFProducer/plugins/importers/GeneralTracksImporter.cc

```diff
@@ -13,33 +14,84 @@ class GeneralTracksImporter : public BlockElementImporterBase {
   GeneralTracksImporter(const edm::ParameterSet& conf, edm::ConsumesCollector& sumes)
       : BlockElementImporterBase(conf, sumes),
         src_(sumes.consumes<reco::PFRecTrackCollection>(conf.getParameter<edm::InputTag>("source"))),
+        vetoEndcap_(conf.getParameter<bool>("vetoEndcap")),
         muons_(sumes.consumes<reco::MuonCollection>(conf.getParameter<edm::InputTag>("muonSrc"))),
         trackQuality_(reco::TrackBase::qualityByName(conf.getParameter<std::string>("trackQuality"))),
-        DPtovPtCut_(conf.getParameter<std::vector<double> >("DPtOverPtCuts_byTrackAlgo")),
-        NHitCut_(conf.getParameter<std::vector<unsigned> >("NHitCuts_byTrackAlgo")),
+        DPtovPtCut_(conf.getParameter<std::vector<double>>("DPtOverPtCuts_byTrackAlgo")),
+        NHitCut_(conf.getParameter<std::vector<unsigned>>("NHitCuts_byTrackAlgo")),
         useIterTracking_(conf.getParameter<bool>("useIterativeTracking")),
         cleanBadConvBrems_(conf.getParameter<bool>("cleanBadConvertedBrems")),
-        muonMaxDPtOPt_(conf.getParameter<double>("muonMaxDPtOPt")) {}
+        muonMaxDPtOPt_(conf.getParameter<double>("muonMaxDPtOPt")) {
+    if (vetoEndcap_) {
+      vetoMode_ = conf.getParameter<unsigned>("vetoMode");
+      switch (vetoMode_) {
+        case 1:
```

Comment:
it may be worth specifying an enum for `vetoMode_` options, to avoid repeated magic numbers and improve clarity

Reply:
I see. I will try.

Reply:
FYI: After the latest commit, enum is used.